loader/src/sources/cdc/api.js

```diff
@@ -0,0 +1,309 @@
+
+      // TODO: better structure open hours information
+      "sunday_hours",
+      "monday_hours",
+      "tuesday_hours",
+      "wednesday_hours",
+      "thursday_hours",
+      "friday_hours",
+      "saturday_hours",
+    ];
+    const meta = {};
+    for (const field of metaFields) {
+      meta[field] = base[field];
+    }
+
+    const productFields = [
+      "ndc",
+      "med_name",
+      "in_stock",
+      "supply_level",
+      "quantity_last_updated",
+    ];
+    const productList = storeItems.map((item) => {
+      const out = {};
+      for (const field of productFields) {
+        out[field] = item[field];
+      }
+      return out;
+    });
+
+    result = {
+      external_ids: [
+        ["vaccines_gov", base.provider_location_guid],
+        storeExternalId,
+      ],
+      name: titleCase(base.loc_name),
+      provider: storeExternalId[0],
+      address_lines: addressLines,
+      city: titleCase(base.loc_admin_city),
+      state: base.loc_admin_state,
+      postal_code: base.loc_admin_zip,
+      info_phone: base.loc_phone,
+      info_url: base.web_address,
+      meta,
+
+      availability: {
+        source: "cdc",
+        checked_at: new Date().toISOString(),
+        valid_at: formatValidAt(productList),
+        available_count: getAvailableCount(productList),
+        available: formatAvailable(productList),
+        products: formatProductTypes(productList),
+      },
+    };
+
+    if (base.longitude && base.latitude) {
+      result.position = {
+        longitude: base.longitude,
+        latitude: base.latitude,
+      };
+    }
+  });
+  return result;
+}
+
+const systemNameRe = {
+  costco: /^Costco/i,
+  cvs: /^CVS/i,
+  kroger: /^Kroger/i,
+  publix: /^Publix/i,
+  rite_aid: /^Rite Aid/i,
+  safeway: /^SAFEWAY/i,
+  sams_club: /^Sams Club/i,
+  walgreens: /^Walgreens/i,
+  walmart: /^Walmart/i,
+};
+
+function getStoreExternalId(store) {
+  // handle numeric store numbers
+  let m = store.loc_store_no.match(/^(?<storeNo>\d+)$/);
+
+  if (!m) {
+    // handle vtrcks pins like RA105587 -> 5587 in addition to pure numeric store numbers
+    m = store.loc_store_no.match(/^([A-Z]{3}|[A-Z]{2}\d)(?<storeNo>\d{5})/i);
+  }
+
+  if (!m) {
+    return null;
+  }
+  const storeNumber = parseInt(m.groups.storeNo, 10).toString();
+
+  for (const system in systemNameRe) {
+    if (store.loc_name.match(systemNameRe[system])) {
+      return [system, storeNumber];
+    }
+  }
+}
+
+function formatValidAt(products) {
+  const dates = products.map((p) => p.quantity_last_updated);
+  dates.sort();
+  return dates[dates.length - 1];
+}
+
+function getAvailableCount(products) {
+  if (!Array.isArray(products)) {
+    products = [products];
+  }
+
+  let sum = 0;
+  for (const product of products) {
+    const supplyLevel = parseInt(product.supply_level, 10);
+    if (supplyLevel > 0) {
+      sum += supplyLevel;
+    } else {
+      sum += product.in_stock ? 1 : 0;
+    }
+  }
+  return sum;
+}
```

Comment:
Welp, I discovered some documentation we must have missed today that explains this, and the `supply_level` column is not an actual count in any way. It’s an enum describing *how long current supply should last*. https://data.cdc.gov/api/views/5jp2-pgaw/files/2a8f085d-04ab-4044-8148-3b7df4cb7088?download=true&filename=Outbound%20File%20Specification.pdf

| Value | Definition |
| ----: | ---------- |
| -1 | No report |
| 0 | No supply |
| 1 | < 24 hours of supply |
| 2 | 24–48 hours of supply |
| 3 | > 24 hours of supply |

…so we should probably only be using this to infer yes/no for `available`, and not try to set `available_count` with it at all.

Reply:
😮 

Will fix, thanks.